atlassian-labs/figma-for-jira
# NONE: Add link to switch users on admin page

This PR adds the following UI in the connect team flow:
![image](https://github.com/atlassian-labs/figma-for-jira/assets/6136959/59e11b1e-de2c-463a-8734-d4d305932ad7)

However, in order to show the email and provide a link to re-initiate the auth flow, we needed to refactor our backend a bit. The changes here are:
- `checkUserFigmaAuthUseCase` -> `currentFigmaUserUseCase` so we're grabbing the relevant details about the logged in figma user (we only keep the email rn since that's all we're showing)
- updating `admin/auth/checkAuth` to always return the authorization endpoint so the user can switch accounts even if they are already logged in
- updating the internals of `/auth/checkAuth` to use `currentFigmaUserUseCase` instead. The response from this endpoint is unchanged

### Test Plan
- open the admin page
- try to connect a team
- see that it displays your logged in email at the bottom
- see that clicking the link takes you to the start of the auth flow
